tsfc/fiatinterface.py

```diff
@@ -178,6 +173,15 @@ def _(element, vector_is_mixed):
         # Real element is just DG0
         cell = element.cell()
         return create_element(ufl.FiniteElement("DG", cell, 0), vector_is_mixed)
+    if element.family() == "Quadrature":
+        # Sneaky import from FFC
+        from ffc.quadratureelement import QuadratureElement
+        ffc_element = QuadratureElement(element)
+
+        def tabulate(self, order, points, entity):
+            return QuadratureElement.tabulate(self, order, points)
+        ffc_element.tabulate = types.MethodType(tabulate, ffc_element)
+        return ffc_element
```

Comment:
It's not really a bug from FFC's point of view, but only from TSFC's point of view.

TSFC always passes "FIAT elements" the points in the reference coordinates of the appropriate subentity, and uses the `entity=` parameter to specify which subentity the coordinates are to be interpreted on. FIAT elements have been updated to handle this entity parameter, and map points from facet to cell coordinates themselves if needed, or use the entity parameter directly (e.g. trace element).

Meanwhile, FFC itself translates quadrature points from facet to cell coordinates, and then asks FIAT elements to always tabulate on the cell, and thus never uses the optional entity parameter. Now, `QuadratureElement` resides in FFC, but its duck type is sort of a FIAT element.